src/node/types.h

```diff
@@ -99,6 +102,35 @@ struct BlockCheckOptions {
     bool check_pow{true};
 };
 
+struct CoinbaseTemplate {
+    /* nVersion */
+    uint32_t version;
+    /* nSequence for the only coinbase transaction input */
+    uint32_t input_sequence;
+    /**
+     * Bytes which are to be placed at the beginning of scriptSig. Guaranteed
+     * to be less than 8 bytes (not including the length byte). This allows
+     * clients to add up to 92 bytes.
+     */
+    CScript script_sig_prefix;
+    /**
+     * The first (and only) witness stack element of the coinbase.
+     *
+     * Omitted for block templates without witness data.
+     *
+     * This is currently the BIP 141 witness reserved value. A future soft fork
+     * may move the witness reserved value elsewhere, but there will still be a
+     * coinbase witness.
```

Comment:
In commit "mining: add getCoinbase()" (80acabaf9353245f0839b18d9552ab39f595eda6)

I'm a little confused by "may move the witness reserved value" in this comment. I would think a future soft fork could restrict this value but not move it anywhere. BIP 141 says the coinbase witness has to be a single 32-byte so that doesn't seem to allow any other flexibility.

Reply:
The BIP says:

> For backward compatibility, the `Hash(new commitment|witness reserved value)` will go to the coinbase witness, and the witness reserved value will be recorded in another location specified by the future softfork. Any number of new commitment could be added in this way.

So we provide clients with the coinbase witness, which is not necessarily the "witness reserved value".